api/v1alpha4/types.go

```diff
@@ -58,6 +58,10 @@ type NetworkSpec struct {
 	// APIServerLB is the configuration for the control-plane load balancer.
 	// +optional
 	APIServerLB LoadBalancerSpec `json:"apiServerLB,omitempty"`
+
+	// PrivateDNSZoneName define the zone name for the Azure Private DNS.
+	// +optional
+	PrivateDNSZoneName string `json:"privateDNSZoneName,omitempty"`
```

Comment:
Do we intend for this field to be mutable after cluster creation? If not, we should enforce immutability via webhook validation.

Reply:
This field should not be mutable after cluster creation, I enforced it. 
Just a question, is there any forbidden dns zone name in azure (I didn't see anything about it in the doc)? 
if yes we should forbid them with the validating webhook too.

Reply:
this is what I found: `The name of the zone, for example, contoso.com. The label and top-level domain (TLD) can contain alphanumeric characters, underscores, and dashes, and must be separated using a period.`